localfs/emptydir_test.go

```diff
@@ -0,0 +1,118 @@
+package localfs
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDirectoryCleanup(t *testing.T) {
+	testDir, err := os.MkdirTemp("/tmp", "dirtest")
+	require.NoError(t, err)
+	defer os.RemoveAll(testDir)
```

Comment:
[NIT] `t.TempDir()` does all this for us IIRC:

```suggestion
	testDir := t.TempDir()
```